---
draft: true 
date: 2024-04-08
categories:
  - Python
comments: true
authors: [Blake]
---
这是去年年底就发生的事情了，至于为啥现在拿出来说，是因为我在浏览仓库的的提交记录时，一个直白、残酷的 commit message 再一次地扎进我的眼里：还原 xxx（我的名字）提交错误。这是我同事提交的 commit，用于回退我之前提交的部分代码。就是这部分代码，导致了这次线上事故...

<!-- more -->
## 背景

我们收到了为某个任务执行提速的需求：并发执行。我们的任务执行引擎是支持协程异步执行的，但是这个任务的业务代码是同步的。有两种方案摆在面前：1. 将任务代码改成异步；2. 引擎新增一个 runner，支持多线程执行任务。 
鉴于时间较为紧张，修改任务代码的工作量和风险要远大新增一个多线程执行的 runner，我们最终选择的第二个方案，这个任务也被我给接下了。

## 开发

新加一个线程 runner 并不复杂，再加上一些锁的处理，半天时间就开发完毕，提交上线了。

## 事故


## __init__subclass__

## 教训



